src/lib/config-db.ts

```diff
@@ -23,14 +23,14 @@ export async function loadConfigFromLocalStorage (): Promise<void> {
   if (typeof globalThis.localStorage !== 'undefined') {
     await configDb.open()
     const localStorage = globalThis.localStorage
-    const localStorageGatewaysString = localStorage.getItem(LOCAL_STORAGE_KEYS.config.gateways) ?? JSON.stringify(defaultGateways)
-    const localStorageRoutersString = localStorage.getItem(LOCAL_STORAGE_KEYS.config.routers) ?? JSON.stringify(defaultRouters)
-    const localStorageDnsResolvers = localStorage.getItem(LOCAL_STORAGE_KEYS.config.dnsJsonResolvers) ?? JSON.stringify(defaultDnsJsonResolvers)
+    const localStorageGatewaysString = localStorage.getItem(LOCAL_STORAGE_KEYS.config.gateways) ?? convertUrlArrayToInput(defaultGateways)
```

Comment:
nit: would it be possible to keep existing behavior (store JSON) and convert to prettier newline delimited representation only in the end users GUI?

Right now, existing users that have JSON in storage are being broken (screenshot below):

> ![2024-07-04_02-06](https://github.com/ipfs/service-worker-gateway/assets/157609/b3ef2beb-c734-40bc-a084-6d84f9c70ac3)

Reply:
totally. let me fiddle with that. i created functions for conversion both ways so that shouldn't be a big deal.

Reply:
ok @lidel give it a try with [add703b](https://github.com/ipfs/service-worker-gateway/pull/315/commits/add703b9e46fc13436ef79498835e1d37323cb33)